cmd/filestransfer/filestransfer.go

```diff
@@ -0,0 +1,180 @@
+// Example: Files transfer using PUT/GET commands
+//
+// This example shows how to transfer files to staging area, from which data can be loaded into snowflake
+// database tables. Apart from sending files to staging area using PUT command, files can also be downloaded
+// using GET command.
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	sf "github.com/snowflakedb/gosnowflake"
+)
+
+const customFormatCsvDataToUpload = "NUM; TEXT\n1; foo\n2; bar\n3; baz"
+
+func getDSN() (string, *sf.Config, error) {
+	env := func(key string, failOnMissing bool) string {
+		if value := os.Getenv(key); value != "" {
+			return value
+		}
+		if failOnMissing {
+			log.Fatalf("%v environment variable not set", key)
+		}
+		return ""
+	}
+	account := env("SNOWFLAKE_TEST_ACCOUNT", true)
+	user := env("SNOWFLAKE_TEST_USER", true)
+	password := env("SNOWFLAKE_TEST_PASSWORD", true)
+	database := env("SNOWFLAKE_TEST_DATABASE", true)
+	warehouse := env("SNOWFLAKE_TEST_WAREHOUSE", true)
+	host := env("SNOWFLAKE_TEST_HOST", false)
+	schema := env("SNOWFLAKE_TEST_SCHEMA", true)
+	portStr := env("SNOWFLAKE_TEST_PORT", false)
+	protocol := env("SNOWFLAKE_TEST_PROTOCOL", false)
+
+	port := 443
+	var err error
+	if len(portStr) > 0 {
+		port, err = strconv.Atoi(portStr)
+		if err != nil {
+			return "", nil, err
+		}
+	}
+
+	cfg := &sf.Config{
+		Account:   account,
+		User:      user,
+		Password:  password,
+		Database:  database,
+		Warehouse: warehouse,
+		Schema:    schema,
+		Host:      host,
+		Port:      port,
+		Protocol:  protocol,
+	}
+
+	dsn, err := sf.DSN(cfg)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return dsn, cfg, nil
+}
+
+func tmpFileWithDataToUpload() string {
+	tempFile, err := os.CreateTemp("", "data_to_upload.csv")
```

Comment:
I'd prefer having folder here. It's easier to locate files and remove them all at once.

Reply:
Does it matter since this is written to OS's tmp directory hence it will be removed automatically?